core/test/matrix/coo.cpp

```diff
@@ -82,10 +91,10 @@ class Coo : public ::testing::Test {
         EXPECT_EQ(c[1], 1);
```

Comment:
Dont these also technically need to be `index_type(1)` and so on ?

Reply:
Well, here since it's a value greatly inferior to `std::numeric_limits<int>::max()`, I believe there is no need for that.

Cf. the relevant gtest code:
https://github.com/google/googletest/blob/23b2a3b1cf803999fb38175f6e9e038a4495c8a5/googletest/include/gtest/gtest.h#L1525-L1536

Reply:
Okay, that makes sense. Thanks.

Reply:
In fact, we are doubly covered because they simply use `operator==` in the code I linked. Here are the rules:
https://en.cppreference.com/w/cpp/language/operator_arithmetic
https://en.cppreference.com/w/cpp/language/implicit_conversion#Integral_promotion

So there really is no need for anything here.